ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/AgentUtils.java

```diff
@@ -295,7 +295,7 @@ public static void parseThoughtResponse(Map<String, String> modelOutput, String
     public static String extractFinalAnswer(String text) {
         String result = null;
         if (text.contains("\"final_answer\"")) {
-            String pattern = "\"final_answer\"\\s*:\\s*\"(.*?)$";
+            String pattern = "\"final_answer\"\\s*:\\s*\"(.*)\"";
```

Comment:
What if the final answer has some " inside it ,for example
```
"final_answer": "This is a \"test answer\""
```

Reply:
It will match `This is a \"test answer\"` since it use `.*` which is a greedy mode instead of `.*?`.

I'll add a new unit test to verify this case.

Reply:
please test os assistant and chat experience, make sure this change will not break it